Promise/Promise.swift

```diff
@@ -86,30 +96,31 @@ enum State<Value>: CustomStringConvertible {
 final class Promise<Value> {
     
     private var state: State<Value>
-    private let lockQueue = dispatch_queue_create("lock_queue", DISPATCH_QUEUE_SERIAL)
+    private let lockQueue = DispatchQueue(label: "promise_lock_queue", qos: .userInitiated)
     private var callbacks: [Callback<Value>] = []
     
     init() {
-        state = .Pending
+        state = .pending
     }
     
     init(value: Value) {
-        state = .Fulfilled(value: value)
+        state = .fulfilled(value: value)
     }
     
-    init(error: ErrorType) {
-        state = .Rejected(error: error)
+    init(error: Error) {
+        state = .rejected(error: error)
     }
     
-    convenience init(queue: dispatch_queue_t = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), work: (fulfill: (Value) -> (), reject: (ErrorType) -> () ) -> ()) {
+    convenience init(queue: DispatchQueue = DispatchQueue.global(qos: .userInitiated), work: @escaping (_ fulfill: @escaping (Value) -> (), _ reject: @escaping (Error) -> () ) -> ()) {
         self.init()
-        dispatch_async(queue, {
-            work(fulfill: self.fulfill, reject: self.reject)
-        })
+        queue.async {
```

Comment:
yeah, i think so. i'm still going back and forth on when to use trailing closure syntax, but i think i'm leaning towards "never"